internal/wait/wait.go

```diff
@@ -41,6 +42,34 @@ func ClusterPolicyReady(apiClient *clients.Settings, clusterPolicyName string, p
 		})
 }
 
+// NicClusterPolicyReady Waits until nicClusterPolicy is Ready.
+func NicClusterPolicyReady(apiClient *clients.Settings, nicClusterPolicyName string, pollInterval,
```

Comment:
I think it's a bad idea to have all waits for all use-cases in the same package. IMO it would make more sense to have respective wait packages for the GPU operator, network operator, NFD operator, etc.

Less room for mistakes, e.g. using `glog.V(gpuparams.GpuLogLevel)` to log for the network operator :grin: 

Functions like `wait.PollUntilContextTimeout()`, on the other hand, can be generic and used by any component.

Reply:
I agree here, I created a networkwait.go file with Network Operator specific code.  We will need to do the same and put the clusterpolicy specific cod in a new gpuwait.go file, when we pull out NFD from both GPU and Network Operators test code.